spring-webflux/src/main/java/org/springframework/web/reactive/function/client/DefaultWebClientBuilder.java

```diff
@@ -335,16 +344,18 @@ public WebClient build() {
 
 	private ClientHttpConnector initConnector() {
 		if (reactorNettyClientPresent) {
-			return new ReactorClientHttpConnector();
+			return new ReactorClientHttpConnector(this.applyAttributes);
 		}
 		else if (reactorNetty2ClientPresent) {
-			return new ReactorNetty2ClientHttpConnector();
+			return new ReactorNetty2ClientHttpConnector(this.applyAttributes);
 		}
 		else if (jettyClientPresent) {
-			return new JettyClientHttpConnector();
+			return new JettyClientHttpConnector(this.applyAttributes);
 		}
 		else if (httpComponentsClientPresent) {
-			return new HttpComponentsClientHttpConnector();
+			HttpComponentsClientHttpConnector httpComponentsClientHttpConnector = new HttpComponentsClientHttpConnector();
+			httpComponentsClientHttpConnector.setApplyAttributes(this.applyAttributes);
```

Comment:
For reasons unknow to me I was not able to call the new constructor with `applyAttributes` parameter. The compile Task threw an error saying:
```
 Task :spring-webflux:compileJava
/tmp/build/f663b219/git-repo/spring-webflux/src/main/java/org/springframework/web/reactive/function/client/DefaultWebClientBuilder.java:356: error: cannot access CloseableHttpAsyncClient
			return new HttpComponentsClientHttpConnector(applyAttributes);
			       ^
  class file for org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient not found
``` 
(see [pipelines/builds/1101](https://ci.spring.io/teams/spring-framework/pipelines/spring-framework-6.0.x/jobs/build-pull-requests/builds/1101)).
Therefore I used the default constructor and the setter.